common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -1363,9 +1363,9 @@ public static UTF8String trimRight(
 
   public static UTF8String[] splitSQL(final UTF8String input, final UTF8String delim,
       final int limit, final int collationId) {
-    if (CollationFactory.fetchCollation(collationId).supportsBinaryEquality) {
+    if (CollationFactory.fetchCollation(collationId).isUtf8BinaryType) {
```

Comment:
I wouldn't recommend completely abandoning the concept of binary equality

UTF8_BINARY is the only collation that supports binary equality at this moment, but we may have others in the future - so we shouldn't equate the two just yet

Reply:
I agree, but I didn't abondon supportsBinaryEquality, it still exists and will be extensively used.
Although there might be places where I used supportsBinaryEquality instead of supportsBinaryEquality. Will fix it:
https://github.com/apache/spark/pull/48472/files#diff-640c14aa5d7473df79b2435ce5a327dffcc16ca29354b153956b4f8d19fdb16c